ocaml/xapi/xapi_sr.ml

```diff
@@ -183,12 +183,44 @@ let probe ~__context ~host ~device_config ~_type ~sm_config =
 
   transform_storage_exn
     (fun () ->
-       match Client.SR.probe ~dbg ~queue ~device_config ~sm_config with
-       | Raw x -> x
-       | Probe _ as x -> Xmlrpc.to_string (rpc_of_probe_result x)
+       Client.SR.probe ~dbg ~queue ~device_config ~sm_config |> f
     )
 
-let probe_ext ~__context ~host ~device_config ~_type ~sm_config =
+let probe = call_probe ~f:(
+    function
+    | Storage_interface.Raw x -> x
+    | Storage_interface.Probe probe_results ->
+      (* Here we try to mimic the XML document structure returned in the Raw
+         case by SMAPIv1, for backwards-compatibility *)
+      let module T = Xmlm_tree in
```

Comment:
Maybe we should add ezxmlm to xs-opam (ezjsonm  is already there) and use that instead?

Reply:
Sounds good. However, it's hard to branch xs-opam. So we should add it to master and remove this next time we rebase.